src/js/mep-player.js

```diff
@@ -850,10 +862,45 @@
 			if (typeof height != 'undefined') {
 				t.height = height;
 			}
-
+ 
+            		// check stretching modes
+            		switch(t.options.stretching) {
+                		case 'fill':
+        				// The 'fill' effect only makes sense on video; for audio we will set the dimensions
+                    			if (t.isVideo) {
+                    				this.setFillMode();
+ 					} else {
+                                		this.setDimensions(t.width, t.height);
+ 					}
+                    			break;
+                		case 'responsive':
+                    			this.setResponsiveMode();
+                    			break;
+                		case 'none':
+                    			this.setDimensions(t.width, t.height);
+                    			break;
+                		// This is the 'auto' mode
+                		default:
+                    			if (this.hasFluidMode() === true) {
+                        			this.setResponsiveMode();
+                    			} else {
+                        			this.setDimensions(t.width, t.height);
+                    			}
+                    			break;
+            		}
+		},
+ 
+	        hasFluidMode: function() {
+	        	var t = this;
+	 
 			// detect 100% mode - use currentStyle for IE since css() doesn't return percentages
-			if (t.height.toString().indexOf('%') > 0 || (t.$node.css('max-width') !== 'none' && t.$node.css('max-width') !== 't.width') || (t.$node[0].currentStyle && t.$node[0].currentStyle.maxWidth === '100%')) {
-
+			return (t.height.toString().indexOf('%') > 0 || (t.$node.css('max-width') !== 'none' && t.$node.css('max-width') !== 't.width') || (t.$node[0].currentStyle && t.$node[0].currentStyle.maxWidth === '100%'));
+	        },
+ 
+	        setResponsiveMode: function() {
+			var t = this;
+			if (t.hasFluidMode) {
```

Comment:
Why are you checking for the existence of the function? This will always be true? 
If I change this to the correct `if (t.hasFluidMode()) {` then it fails pretty hard if it's false. This means you need to do both stretching: 'responsive' and use `max-width:100%`